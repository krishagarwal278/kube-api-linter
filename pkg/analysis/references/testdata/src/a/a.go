package a

type TestSpec struct {
	NodeReference string `json:"nodeReference"` // want "field NodeReference should use 'Ref' instead of 'Reference'"

	NodeReferences []string `json:"nodeReferences"` // want "field NodeReferences should use 'Refs' instead of 'References'"

	// AllowedRef field should be valid
	AllowedRef string `json:"allowedRef"`

	// AllowedRefs field should be valid
	AllowedRefs []string `json:"allowedRefs"`

	// ValidField should be valid
	ValidField string `json:"validField"`

	// ParentRef should be valid
	ParentRef string `json:"parentRef"`

	// ParentRefs should be valid
	ParentRefs []string `json:"parentRefs"`
}
