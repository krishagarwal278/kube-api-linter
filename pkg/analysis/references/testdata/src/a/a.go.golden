package a

type TestSpec struct {
	NodeRef string `json:"nodeReference"` // want "field NodeReference should use 'Ref' instead of 'Reference'"

	NodeRefs []string `json:"nodeReferences"` // want "field NodeReferences should use 'Refs' instead of 'References'"

	// AllowedRef field should be valid
	AllowedRef string `json:"allowedRef"`

	// AllowedRefs field should be valid
	AllowedRefs []string `json:"allowedRefs"`

	// ValidField should be valid
	ValidField string `json:"validField"`

	// ParentRef should be valid
	ParentRef string `json:"parentRef"`

	// ParentRefs should be valid
	ParentRefs []string `json:"parentRefs"`
}
